model/src/main/kotlin/OrtResult.kt

```diff
@@ -110,7 +110,7 @@ data class OrtResult(
         sortedMapOf<String, SortedSet<Identifier>>().also { licenses ->
 
             getProjects().forEach { project ->
-                if (!omitExcluded || !isProjectExcluded(project)) {
+                if (!omitExcluded || !isProjectExcluded(project.id)) {
```

Comment:
Hmm. What I don't like so much about this change is that with the new signature you can accidentally pass in a package's `id`. And in general, for a function called "is*Project*Excluded()", I find it more fitting to take a `Project` as the parameter. Shouldn't we rather change `isPackageExcluded()` to take a `Package`?

Reply:
Agree that this signature change introduces the risk that the caller passes any other ID.
But even if accidentally done this would not introduce a bug into the caller considering the semantics of that function.

An argument for this change imo is that it reduces dependencies which in general is
good also in terms of testability. Also it is now consistent with isPackageExcluded which I think is good. For the consistency it is worth mentioning that changing instead isPackageExcluded to accept packages is not an easy thing to do as that function also checks for project exclusion.

Given the above for me it is not 100% clear which one is better, I have to say that I tend to agree with "not changing any signature of those two function at all in this PR" I propose to keep it as and due to the time pressure I have for merging this and as I think the PR overall makes the codebase better than before.

I'd promise to create a follow up PR to address your doubts.
I'd also like to experiment with change isProjectExcluded to isProjectIncluded and see how much that would simplify the callers code. That seems to have potential as well. If this was done there would be less doubts about the ID parameter I assume.

Are you fine with that?

Reply:
> An argument for this change imo is that it reduces dependencies which in general is
good also in terms of testability.

I'm a tad confused about what you call "dependencies" here. Do you mean that the function does not "depend" on the `Project` class anymore, as it's directly getting an `Identifier` instead of an `Identifier` "nested" in a `Project`? If that's what you mean, how does that fact really help with anything?

> Also it is now consistent with isPackageExcluded which I think is good.

For sure consistency is a good thing. And as I trust you that you've investigated that instead changing `isPackageExcluded()` to take a `Package` is less feasibly, I'm fine with the change.

> I'd also like to experiment with change isProjectExcluded to isProjectIncluded and see how much that would simplify the callers code.

Interesting idea to try (later) :+1: 

> Are you fine with that?

Yes, thanks.

Reply:
Allright, thanks for your input! I'll follow up on this with a future PR then.

> I'm a tad confused about what you call "dependencies"

yeah its minor - a test case would just need to construct an identifier for calling that function (under test) which might be less bloated and more readable compared to constructing a project.